django/forms/models.py

```diff
@@ -1112,7 +1112,7 @@ def __iter__(self):
             yield self.choice(obj)
 
     def __len__(self):
-        return (len(self.queryset) + (1 if self.field.empty_label is not None else 0))
+        return (len(self.queryset.all()) + (1 if self.field.empty_label is not None else 0))
```

Comment:
In which cases does `__len__` gets called by the form layer? I'm asking because if it's only to allow third party to do `if field.choices` we better be implementing an optimized `__bool__` as well because `.count()` can perform bad on large resultsets.

```python
def __bool__(self):
    return self.field.empty_label is not None or self.queryset.exists()
```